core/arch/arm/plat-k3/drivers/ti_sci.c

```diff
@@ -425,6 +425,31 @@ int ti_sci_lock_otp_row(uint8_t row_idx, uint8_t hw_write_lock,
 	return 0;
 }
 
+int ti_sci_set_swrev(uint8_t identifier, uint32_t swrev)
+{
+	struct ti_sci_msq_req_set_swrev req = { };
+	struct ti_sci_msq_resp_set_swrev resp = { };
+	struct ti_sci_xfer xfer = { };
+	int ret = 0;
+
+	ret = ti_sci_setup_xfer(TI_SCI_MSG_WRITE_SWREV, 0,
+				&req, sizeof(req),
+				&resp, sizeof(resp),
+				&xfer);
+	if (ret)
+		return ret;
```

Comment:
The change looks good to me in that it is consistent with the rest if **ti_sci.c** implementation. That said, I see that all the functions in this source file use `int` return values whereas they return `TEE_ERROR_xxx` or `0` values. I think it would be nice someone changes this to use `TEE_Result` explicit return values and `TEE_SUCCESS` instead of `0` where applicable.

Reply:
Agree, this has been an issue for a while, will be good to fix this all these functions to return `TEE_Result` after we finish the churn of this series.